### DISCUSSION TITLE
Passing state to navigate

### DISCUSSION THREAD

**Discussion by cwagner22** [vikejs/vike#1976] [Discussion]
Is there any way to pass state to `navigate()` like `react-router-dom`?

**Comment by brillout** [comment]
What's your use case?

**Reply by cwagner22** [reply]
I have a login form and if "Forget Password" is pressed is would like to navigate to the reset password page with the email input filled of the previous value entered. 
I was also using the navigate state when the user was accessing a private url not logged in, passing the original url as param. But I will find a way to store it before I call redirect().
I'm more blocked by the lack of [modals route support](https://github.com/vikejs/vike/issues/1862).

**Reply by brillout** [reply]
[Done](https://github.com/vikejs/vike/commit/9a97c22f50d4250e3fa7a9ce48459c1e76328101) and pre-released as `0.4.204-commit-9a97c22`.

> I'm more blocked by the lack of [modals route support](https://github.com/vikejs/vike/issues/1862).

In general I can see use cases where modal route interception makes sense. Although for your use case, it seems like simply rendering a modal (without changing the URL) would be better? The more I understand how the lack of route interception is a blocker the better. Alternatively, you can use `throw render()` but I guess you want to show the page in a modal.
